frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/AddDataModal/Panels/CSVPanel.tsx

```diff
@@ -0,0 +1,43 @@
+import { t } from "ttag";
+
+import * as Urls from "metabase/lib/urls";
+
+import { CSVPanelEmptyState } from "./AddDataModalEmptyStates";
+import { CSVUpload } from "./CSVUpload";
+
+interface CSVPanelProps {
+  canUpload: boolean;
+  canManageUploads: boolean;
+  onCloseAddDataModal: () => void;
+  uploadsEnabled: boolean;
+}
+
+export const CSVPanel = ({
+  canUpload,
+  canManageUploads,
+  onCloseAddDataModal,
+  uploadsEnabled,
+}: CSVPanelProps) => {
+  return (
+    <>
+      {uploadsEnabled &&
+        (canUpload ? (
+          <CSVUpload onCloseAddDataModal={onCloseAddDataModal} />
+        ) : (
+          <CSVPanelEmptyState contactAdminReason="obtain-csv-upload-permission" />
+        ))}
+
+      {!uploadsEnabled &&
+        (canManageUploads ? (
+          <CSVPanelEmptyState
+            ctaLink={{
+              text: t`Enable uploads`,
+              to: Urls.uploadsSettings(),
+            }}
+          />
+        ) : (
+          <CSVPanelEmptyState contactAdminReason="enable-csv-upload" />
+        ))}
+    </>
+  );
```

Comment:
nit: I find nested conditionals in return statements kinda hard to parse. one pattern that is helpful is having multiple returns, maybe the first return could handle the various empty states, and then the fallback could handle the actual upload component?

```tsx

if (!uploadsEnabled || !canUpload) {
  return <CSVPanelEmptyState ...  />;
}

return <CSVUpload onCloseAddDataModal={onCloseAddDataModal} />

Reply:
In general, I absolutely agree and I tend to avoid nested conditionals as much as possible.
In this particular case, the logic looked solid to me because it flows nicely:

Two separate groups:
1. uploads enabled
2. uploads disabled

And then within them the logic is - can you do something with it or not.

I'll see if I can make this even more straight-forward.

Reply:
This should be easier to read.
https://github.com/metabase/metabase/pull/59008/commits/55cf21834bfb6a11aaa40df285d7bb323c98170a

Reply:
that is nicer, but one thing that's even nicer about multiple returns is you don't have to reason about whether multiple conditions can be true.